test/e2e/autoscaling/cluster_size_autoscaling.go

```diff
@@ -795,6 +795,53 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
 		Expect(getNAPNodePoolsNumber()).Should(Equal(0))
 		framework.ExpectNoError(disableAutoprovisioning())
 	})
+
+	It("shouldn't add new node group if not needed [Feature:ClusterSizeAutoscalingScaleWithNAP]", func() {
+		framework.SkipUnlessProviderIs("gke")
+		By("Decrease size of groups")
+		nodesCountAfterResize := 0
+		for mig, size := range originalSizes {
+			if size > 1 {
+				framework.ExpectNoError(framework.ResizeGroup(mig, int32(1)))
+				nodesCountAfterResize++
+			}
+		}
+		framework.ExpectNoError(framework.WaitForReadyNodes(c, nodesCountAfterResize, resizeTimeout))
+		Expect(getNAPNodePoolsNumber()).Should(Equal(0))
+		framework.ExpectNoError(enableAutoprovisioning(""))
+		By("Create pods")
+		ReserveMemory(f, "memory-reservation", nodesCountAfterResize+1, int(float64(nodesCountAfterResize+1)*float64(0.7)*float64(memAllocatableMb)), true, scaleUpTimeout)
```

Comment:
If I understand correctly:
- nodesCountAfterResize+1 replicas correspond to one pod per node (with intended scale up by 1),
- each of the pre-exisitng nodesCountAfterResize nodes is in a different node group,
- we assume here each of these nodes has allocatable memory equal to memAllocatableMb,
- each of these nodes will have 0.7 * memAllocatableMb free memory.

This doesn't take into account:
- system pods taking over 0.3 * memAllocatableMb on any of the nodes (which is unlikely as the initial state, but not impossible with all the shuffling pods around that happens in this test suite),
- node groups in different groups may have different machine types (and so different allocatable memory value.)

Reply:
Changed expected number of nodes created to more or equal to 1.